\section{Results}
\subsection{Automatic generation of PH model from the PID network}
%give statistics
PH models are written using the PINT\footnote{\url{http://process.hitting.free.fr}} format.  
PINT implements stochastic simulations and static analyses for computing dynamical properties on very large-scale PH models.
For the PINT code generation two procedures were used.
The first procedure detects motifs (controller and controlled components) in graphs from the Pathway Interaction Database; the second, 
generates the PINT code by choosing an adequate concurrency rule, based on synchronization sorts, to represent the motif dynamic in PH.
With this method it is possible to convert the whole content of the PID database into a PH model, as well as individual pre-selected pathways, 
as is the case for the system under study.  It is implemented in Java and available upon request.

\subsection{Simulation of Calcium stimulated biological system}
We simulated the model with and without the inclusion of the synchronization sort. In the following, we present the results of 
the simulation.

\subsubsection{Without the introduction of the synchronization sort.}
One can notice in \pref{fig:rwos} the occurrence of oscillations. Whereas it is not the expected behavior from the biological system,
 it is coherent with the choice of the modeling and the way the simulator works as explained in Section \ref{sssec:synchronization}.
In this simulation, cooperation sorts were used to model multiple controllers of a common controlled (target) component.
It is important to notice that the intensity of the oscillation is linked with 
the size of the concurrence, i.e. the number of controllers a controlled component has.
Despite the presence of the oscillations, the model reproduces expected dynamical behaviors  namely
the dynamics of components, the signal transduction and takes into account the stochastic and time aspect of the model.

\begin{figure}[H]
\centering
\includegraphics[width=5in,height=4in]{images/resultWOS.pdf}
\caption{{\bf Results of system simulations without introducing the synchronization sort for 9 genes.} The 
traces representing the discretized time-series data are shown as black lines.  The traces representing the simulated traces are shown as blue lines.}
\label{fig:rwos}
\end{figure}

\subsubsection{With the introduction of the synchronization sort.}
In \pref{fig:rws} we can see that the introduction of the synchronization sort significantly reduces the 
impact of concurrency. The result shows  a 
clear elimination of the previously observed oscillations (\pref{fig:rwos}). 
Comparing the simulated results with the ones observed experimentally, we found four different cases.
We found $5$ simulation traces (IL8, uPAR, IL1\_beta, ET1, A20) that matched the sequence of all the component expression levels perfectly.  In this case, delays exist among simulation and experiment but 
these delays are not comparable since experimental time-points are measured in hours and simulation-units for the simulated PH model.
We found $6$ simulation traces (MKP1, MKP3, Hes5, SM22, TfR, DKK1) that matched the sequence of experimental discrete expression levels missing one expression-level.
We found $1$ components (TNF-alpha) in which at least 2 expression levels are missed.

\begin{figure}[H]
\centering
\includegraphics[width=5in,height=7in]{images/12genes_sim.png}
\caption{{\bf Results of simulations by introducing the synchronization sort.} The gray traces represent the experimental expected behaviors
from the discretization of the time-series data. The blue traces show the simulated behavior.}
\label{fig:rws}
\end{figure}


%%%%%images des noeuds clÃ©s

\subsubsection{Simulating biological processes.}
To validate our model, we studied the prediction of non-observed components of such a system and we focused on biological processes linked 
to Calcium stimulation, such as keratinocyte-differentiation, cell-adhesion and cell-cycle arrest.
Our results are shown in \pref{fig:knodes} and confirm literature experimental evidences on these processes.
In the case of keratinocyte-differentiation, this was a functional behavior measured on the cultured cells upon Calcium stimulation,
 so there was experimental evidences of this effect before measuring the gene expression.  
In the case of cell-cycle arrest, the switch-on of this component represents the fact that the E-cadherin stimulated model predicts the stop 
of growth, as confirmed by literature in human and mouse keratinocytes~\cite{Kolly2005}.
Finally, the cell-adhesion component is predicted to switch-on, also in according to published evidence~\cite{Tu2011} in human and mouse keratinocytes.

\begin{figure}[H]
\centering
\includegraphics[width=4in,height=3.5in]{images/key_nodes1.png}
\caption{{\bf Results of the prediction of biological processes.} The gray traces represent the experimental and literature-based evidence.
The blue traces show the simulated behavior of E-cadherin and three biological processes.}
\label{fig:knodes}
\end{figure}

\subsection{Model validation: traces analysis}

To validate the results of the simulations, we automatically analyzed the traces generated by a set of $100$ simulations. Table \ref{traceAnalysis} shows the results
of the percentage of acceptance for the traces of each of the $12$ mRNA expressions. One can observe that there are $4$ components with a good 
acceptance rate ($> 76\%$), which are: A20, IL1$\_$beta, IL8, uPar;
$4$ traces with a good acceptance rate ($> 94\%$) when considering $1$ level of tolerance, which are: MKP1, MKP3, SM22, and TfR;
and finally $4$ traces, for which the model failed to predict their expressions:  ET1, Hes5, DKK1, and TNFa.
All in all, for this case study our model predicts relatively well, $8$ out of $12$, the experimental traces.
Errors on the prediction of the missing $4$ components may be because of missing regulatory interactions.

\begin{table}[!t]
\renewcommand{\arraystretch}{1.3}


\caption{Percentage of acceptance traces. First column represents the Automaton ($\mathcal{A}_{i}(w)$, where $\mathcal{A}_{i}$ is the Automaton and $w$ is the word recognized by $\mathcal{A}_{i}$) 
that is used to check if a given trace is accepted for a component in the second 
column. One can observe that many components can be recognized by the same Automaton. 
In the third column we show the percentage of accepted traces; in the fourth column, 
 the percentage of acceptance with a tolerance of one level ($T_{1}$).}


\label{traceAnalysis}
\centering
%table of result
\begin{tabular}{|c|c||c|c|}
\hline

\textbf{Automate}

&

\textbf{components}

&

\textbf{$\%$ of acceptance}

&

\textbf{$\%$ of acceptance $T_{1}$}
\\ \hline

$\mathcal{A}_{2}(01210)$

&

A20

&

91

&

100
\\ \hline

$\mathcal{A}_{2}(01210)$

&

IL1$\_$beta

&

81

&

100
\\ \hline

$\mathcal{A}_{2}(01210)$

&

IL8

&

93

&

100

\\ \hline

$\mathcal{A}_{2}(01210)$

&

TNF$\_$alpha

&

0

&

0

\\ \hline

$\mathcal{A}_{3}(01211)$

&

uPar

&

76

&

99

\\ \hline

$\mathcal{A}_{3}(01211)$

&

ET1

&

8

&

19

\\ \hline

$\mathcal{A}_{4}(0121210)$

&

DKK1

&

13

&

43

\\ \hline

$\mathcal{A}_{5}(0121211)$

&

Hes5


&

0

&

17

\\ \hline

$\mathcal{A}_{5}(0121211)$

&

MKP1


&

9

&

97

\\ \hline

$\mathcal{A}_{6}(0212)$

&

SM22


&

11

&

100

\\ \hline

$\mathcal{A}_{7}(02010)$

&

MKP3


&

11

&

98

\\ \hline

$\mathcal{A}_{8}(02121)$

&

Tfr


&

0

&

94

\\ \hline

 
\end{tabular}
\end{table}
